src/freenet/node/Node.java

```diff
@@ -2539,6 +2533,46 @@ public void realRun() {
 		System.out.println("Node constructor completed");
 	}
 
+	// peers-offers/*.fref files
+	private void processPeersOffersFrefFiles(SubConfig nodeConfig, int configOptionSortOrder) {
+		nodeConfig.register("peersOffersDismissed", false, configOptionSortOrder, true, true,
+				"Node.peersOffersDismissed", "Node.peersOffersDismissedLong", new BooleanCallback() {
+
+					@Override
+					public Boolean get() {
+						synchronized(Node.this) {
```

Comment:
Please don’t synchronize on the `Node` itself, this would allow any plugin to block access to this.